src/Lecture4.hs

```diff
@@ -100,8 +103,14 @@ module Lecture4
     , printProductStats
     ) where
 
-import Data.List.NonEmpty (NonEmpty (..))
+import Control.Monad (guard)
+import Data.Char (isSpace)
+import Data.List (foldl1')
+import Data.List.NonEmpty (NonEmpty (..), map)
+import Data.Maybe (isNothing, fromJust, mapMaybe)
 import Data.Semigroup (Max (..), Min (..), Semigroup (..), Sum (..))
+import Prelude 
```

Comment:
`Prelude` is imported implicitly so you don't need to import it unless you want to hide something from it ot use qualified names 🙂